src/main/java/com/amazonaws/kinesisvideo/demoapp/activity/WebRtcActivity.java

```diff
@@ -259,6 +265,39 @@ private boolean isValidClient() {
         return client != null && client.isOpen();
     }
 
+    private void handlePendingIceCandidates(String clientId) {
+        // Check if ICE candidates are available in the queue and add the candidate
+        while(!pendingIceCandidatesQueue.isEmpty()) {
+            final Event message = pendingIceCandidatesQueue.peek();
+            final IceCandidate iceCandidate = Event.parseIceCandidate(message);
+            if(clientId == message.getSenderClientId()) {
+                final boolean addIce = localPeer.addIceCandidate(iceCandidate);
+                Log.d(TAG, "Added ice candidate after sdp exchange " + iceCandidate + " " + (addIce ? "Successfully" : "Failed"));
```

Comment:
/nit SDP to be consistent for log scraping

Reply:
Done. I remove the client ID entry from peerConnectionFound map as well after submitting the pending candidates.